internal/bzlmod/go_deps.bzl

```diff
@@ -726,6 +732,32 @@ Mismatch between versions requested for Go module {module}:
 
         go_repository(**go_repository_args)
 
+    # Build lookup table for tool_targets
+    importpath_to_repo = {
+        path: module.repo_name
+        for path, module in module_resolutions.items()
+    }
+
+    tool_targets = {}
+    for tool_path in all_tools:
+        segments = tool_path.split("/")
+        module_path = None
+
+        # Start from the longest possible prefix and work down
+        for i in range(len(segments), 0, -1):
+            candidate = "/".join(segments[:i])
+            if candidate in importpath_to_repo:
+                module_path = candidate
+                break
+
+        if module_path:
+            tool_name = segments[-1]
```

Comment:
Could you pick the last segment here that doesn't satisfy a condition such `len(s) >= 2 and s[0] == "v" and s[1:].isdigit()`? We would want to match the Go spec for tool names as closely as possible, but I think that we can ignore collisions for now.